src/home/homed-home.c

```diff
@@ -2593,6 +2593,11 @@ static int on_home_ref_eof(sd_event_source *s, int fd, uint32_t revents, void *u
         if (h->ref_event_source_dont_suspend || h->ref_event_source_please_suspend)
                 return 0;
 
+        if (home_get_state(h) == HOME_INACTIVE) {
```

Comment:
So this looks good, but I think to be fully correct we should also cover the other inactive states here.

i.e. any chance you can add a helper HOME_STATE_IS_INACTIVE() that lists other relavant tests too. Use the existing HOME_STATE_IS_ACTIVE() as blueprint, place the new helper next to it. It probably has to list the following states:

HOME_UNFIXATED, HOME_ABSENT, HOME_INACTIVE, HOME_DIRTY